File: medical_fe/src/services/departmentService.js
import api from "../core/api";

const departmentService = {
    getAll: () => api.get("/departments"),
    create: (data) => api.post("/departments", data),
    update: (id, data) => api.put(`/departments/${id}`, data),
    delete: (id) => api.delete(`/departments/${id}`)
};

export default departmentService;

File: medical_management_BE/src/main/java/com/nhom2/qnu/service/DepartmentService.java
package com.nhom2.qnu.service;

import com.nhom2.qnu.model.Department;
import com.nhom2.qnu.payload.response.DepartmentResponse;
import com.nhom2.qnu.payload.response.DoctorResponse;

import java.util.List;

public interface DepartmentService {
    List<Department> getAll();

    Department create(Department department);

    Department update(String id, Department department);

    void delete(String id);
}
